From 2a95465d30a946247e05457d17681ce20276cf2a Mon Sep 17 00:00:00 2001
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Thu, 11 Jun 2015 16:03:51 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Add mutex lock while freeing
 GEM obj

For DSI command mode, lock the mutex for freeing GEM object
and later unlock it.

Change-Id: If0da22fb05567fdf9a65edd3841e9e18924959b8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7482
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 2ad10dd..682e031 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -719,6 +719,7 @@ static void intel_dsi_post_disable(struct intel_encoder *encoder)
 {
 	struct drm_i915_private *dev_priv = encoder->base.dev->dev_private;
 	struct intel_dsi *intel_dsi = enc_to_intel_dsi(&encoder->base);
+	struct drm_device *dev = encoder->base.dev;
 	u32 val;
 
 	DRM_DEBUG_KMS("\n");
@@ -746,7 +747,9 @@ static void intel_dsi_post_disable(struct intel_encoder *encoder)
 	if (intel_dsi->gem_obj != NULL) {
 		kunmap(intel_dsi->cmd_buff);
 		i915_gem_object_ggtt_unpin(intel_dsi->gem_obj);
+		mutex_lock(&dev->struct_mutex);
 		drm_gem_object_unreference(&intel_dsi->gem_obj->base);
+		mutex_unlock(&dev->struct_mutex);
 	}
 }
 
-- 
1.9.1

